authors/apps/authentication/views.py

```diff
@@ -71,3 +76,59 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordResetAPIView(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = PasswordResetSerializer
+
+    def post(self, request):
+        recipient = request.data.get('email', {})  # user enters email
+        if not recipient:
+            return Response({"message": "Please provide an email address"},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        # generate a new token for email entered
+        token = jwt.encode({"email": recipient},
+                           SECRET_KEY, algorithm='HS256')
+        # confirm if user exists
+        user_exists = User.objects.filter(email=recipient).exists()
+        if user_exists:
+            result = send_email(recipient, token, request)
+            return Response(result, status=status.HTTP_200_OK)
+        else:  # if user does not exist
+            result = {
+                'message': 'Sorry, user with this email does not exist'
+            }
+            return Response(result, status=status.HTTP_404_NOT_FOUND)
+
+
+class PasswordUpdateAPIView(UpdateAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = PasswordResetSerializer
+
+    def update(self, request, *args, **kwargs):
+        token = self.kwargs.get('token')
+        password = request.data.get('password')
```

Comment:
Hi, @sam-karis here I am getting the password from the user and comparing if the confirm_password matches.